tests/integrations/base_tester.py

```diff
@@ -0,0 +1,245 @@
+the commands (i.e. train, export, deploy).
+B. {Integration_name}_tester.py file containing your class implementation with the
+contents outlined below (#1-3)
+C. The tests themselves, implemented in the same class as B
+D. .yaml files specifying which scenarios to run (one per file)
+
+Components needed for an integration-specific test class. More in depth information
+can be found in the code (marked by the item #).
+1. A mapping of commands (i.e. train, export, and deploy) to their CLI command stubs
+2. A mapping of commands to the pydantic models defining their args
+3. Overrides of functions as needed
+"""
+
+import os
+import shutil
+import subprocess
+from multiprocessing.sharedctypes import Value
+from typing import Dict, List, Tuple, Union
+
+import pytest
+import yaml
+from pydantic import BaseModel
+
+from tests.integrations.base_args import (
+    DummyDeployArgs,
+    DummyExportArgs,
+    DummyTrainArgs,
+)
+from tests.integrations.helpers import get_configs_with_cadence
+
+
+pytest.mark.parametrize(
+    "config_path",
+    get_configs_with_cadence(os.environ["NM_TEST_CADENCE"], indirect=True),
+)
+
+
+class BaseIntegrationTester:
+    # 1 The command stubs should not include tasks. To modify the command stubs to add
+    # tasks (or make any other changes), ovveride the get_base_commands() method.
+    command_stubs = {
+        "train": "sparseml.foo_integration.train",
+        "export": "sparseml.foo_integration.export",
+        "deploy": "sparseml.foo_integration.deploy",
+    }
+    # 2 Based on the CLI arguments that each command can accept, a pydantic model should
+    # be created to enforce argument validation and provide easy reference to possible
+    # arg values for devs
+    command_args_classes = {
+        "train": DummyTrainArgs,
+        "export": DummyExportArgs,
+        "deploy": DummyDeployArgs,
+    }
+
+    @pytest.fixture(scope="class")
+    def setup(self, config_path):
+        # A path for a config file which mathes our integration and cadence is provided
+        # The raw config can contain multiple command stages (train, export, deploy)
+        raw_config = yaml.safe_load(config_path)
+        # Remove cadence for easier processing. It's saved, but not utilized later
+        self.cadence = raw_config.pop("cadence")
+
+        # Will be used in tests to skip test if type doesn't apply
+        self.command_types = [_type for _type in raw_config]
+
+        # 3 The command stub is fetched. See method description
+        self.command_stubs_final = self.get_base_commands(raw_config)
+
+        # 3 extract test args.
+        self.test_args = self.get_test_args(raw_config)
+
+        # Command-specific args are loaded into their respective pydantic classes.
+        # pre_args are CLI commands that can precede the python call
+        # e.g. "torch distributed launch" or env variable setting. If any are present,
+        # they are read in as well.
+        self.configs = {
+            _type: {
```

Comment:
```suggestion
            command_type: {
```

Reply:
(there are few more places where this change should be applied, but I will leave them out for now)